sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -347,13 +361,27 @@ public EncryptedBlobClientBuilder credential(StorageSharedKeyCredential credenti
         return this;
     }
 
+    /**
+     * Sets the {@link AzureNamedKeyCredential} used to authorize requests sent to the service.
+     *
+     * @param credential {@link AzureNamedKeyCredential}.
+     * @return the updated EncryptedBlobClientBuilder
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    @Override
+    public EncryptedBlobClientBuilder credential(AzureNamedKeyCredential credential) {
```

Comment:
Looks like most of the storage clients were missing the `AzureNamedKeyCredential` support before. It looks like other languages are missing this too. Could you please file a bug to add support to this?

Reply:
As far as I remember arch board where these credential types were discussed the decision was to leave storage as is in each language unless there's strong reason to do this.
In our case introduction of a trait would be such reason, i.e. why Storage would be not compliant with the trait. Such reason doesn't exist in other languages afaik.